openml/datasets/dataset.py

```diff
@@ -509,7 +510,12 @@ def _encode_if_category(column):
                 )
         elif array_format == "dataframe":
             if scipy.sparse.issparse(data):
-                return pd.SparseDataFrame(data, columns=attribute_names)
+                # SparseDataFrame removed in pandas 1.0.0
+                if LooseVersion(pd.__version__) >= "1.0.0":
```

Comment:
I don't think this is necessary here. If there are ever problems, we should place such checks in a central place, for example the main `__init__.py`.